cpp/src/neighbors/detail/vamana/priority_queue.cuh

```diff
@@ -0,0 +1,384 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _PQ_H_
+#define _PQ_H_
```

Comment:
A small nitpick before the reviews come through: could you please make sure all header files have `#pragma once` at the top and the copyright years show only `2024` (as far as I understand, these all new files, right?)?

Reply:
Yep, add new files. Fixed header dates and added pragma where needed.